elide-core/src/main/java/com/yahoo/elide/core/CancelSession.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+import org.hibernate.Session;
+
+/**
+ * Cancel Session implementation.
+ */
+
+public abstract class CancelSession {
```

Comment:
Instead of creating an abstract class - create a functional interface (with no dependencies on things like hibernate).

Reply:
I think the functionalInterface will require an entityManager as an argument - and as such - it should probably live in the Aggregation Store and JpaStore respectively.